core/src/transport/upgrade.rs

```diff
@@ -77,8 +80,8 @@ where
     T::Error: 'static,
 {
     /// Creates a `Builder` over the given (base) `Transport`.
-    pub fn new(transport: T) -> Builder<T> {
-        Builder { inner: transport }
+    pub fn new(inner: T, version: upgrade::Version) -> Builder<T> {
```

Comment:
I'm not so much of a fan of this. I think there should be a default version that the user can override.
However I see that the builder would have to be rewritten to account for this, so it's fine.

Reply:
There is a default version - `Version` even implements `Default`. Would you prefer having two different variants of `Transport::upgrade`, one without a `Version` parameter and one with such a parameter, whereby the former delegates to `Builder::new(...)` with the default version? If so, any name suggestions for the `upgrade` variant taking a version? `upgrade_with`? `upgrade_using`? `upgrade_with_protocol`? As you see my reason for not doing so was that I struggled with one of the hardest problems in computer science: naming things.